LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/IJWTParser.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities;
+
+import java.text.ParseException;
+import java.util.Map;
+
+/**
+ * A JWT Parser that parses a JWT and returns the claims.
+ */
+public interface IJWTParser {
+
+    /**
+     * Parse a JWT and returns the claims inside the token.
+     *
+     * @param jwt the JWT to parse
+     * @return a Map containing the JWT claims
+     */
+    Map<String, ?> parseJWT(String jwt) throws ParseException;
```

Comment:
I should change this to a custom exception...will do that...and different implementations then can parse their exception to this custom exception...in the case of nimbus, we will map Parse Exception to custom exception...for something else it would be something else mapping to our Custom Exception

Reply:
You can always throw Runtime.

Reply:
Removed `ParseException` from method signature and just throwing a RuntimeException now: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1389/commits/17f30368410c25fc420f8f8804b93927b8c151d2